lib/fa2/common/tzip12.datatypes.jsligo

```diff
@@ -0,0 +1,37 @@
+export type tokenMetadataData = {
+    token_id: nat,
+    token_info: map<string, bytes>
+};
+
+export type tokenMetadata = big_map<nat, tokenMetadataData>;
+
+// Transfer entrypoint
+
+export type atomic_trans = // @layout comb
+ { to_: address, token_id: nat, amount: nat };
+
+export type transfer_from = { from_: address, txs: list<atomic_trans> };
+
+export type transfer = list<transfer_from>;
+
+// Balance_of entrypoint
```

Comment:
Why not put stuff inside namespace Balance_of ? same for each other datatype.

Reply:
I don't get it ? 
Each entrypoint has to defined its own types

Reply:
```
export namespace BalanceOf  {
export type ... 
// all type relatives to BalanceOf
}
```

Goal is to make the library ordered, instead to put every type in root of the namespace NFT